server/routes/auth.ts

```diff
@@ -133,6 +147,120 @@ authRoutes.post('/login', async (req, res, next) => {
+      settings.jellyfin.hostname != ''
+        ? settings.jellyfin.hostname
+        : body.hostname;
+    // First we need to attempt to log the user in to jellyfin
+    const jellyfinserver = new JellyfinAPI(hostname ?? '');
+    settings.jellyfin.name = await jellyfinserver.getServerName();
+
+    const account = await jellyfinserver.login(body.username, body.password);
+
+    // Next let's see if the user already exists
+    let user = await userRepository.findOne({
+      where: { jellyfinId: account.User.Id },
+    });
+
+    if (user) {
+      // Let's check if their authtoken is up to date
+      if (user.jellyfinAuthToken !== account.AccessToken) {
+        user.jellyfinAuthToken = account.AccessToken;
+      }
+
+      // Update the users avatar with their jellyfin profile pic (incase it changed)
+      if (typeof account.User.PrimaryImageTag !== undefined) {
+        user.avatar = `${hostname}/Users/${account.User.Id}/Images/Primary/?tag=${account.User.PrimaryImageTag}&quality=90`;
+      } else {
+        user.avatar = '/images/os_logo_square.png';
+      }
+      user.email = account.User.Name;
+      user.jellyfinUsername = account.User.Name;
+
+      if (user.username === account.User.Name) {
+        user.username = '';
+      }
+      await userRepository.save(user);
+    } else {
+      // Here we check if it's the first user. If it is, we create the user with no check
+      // and give them admin permissions
+      const totalUsers = await userRepository.count();
+
+      if (totalUsers === 0) {
+        user = new User({
+          email: account.User.Name,
+          jellyfinUsername: account.User.Name,
+          jellyfinId: account.User.Id,
+          jellyfinAuthToken: account.AccessToken,
+          permissions: Permission.ADMIN,
+          avatar:
+            typeof account.User.PrimaryImageTag !== undefined
+              ? `${hostname}/Users/${account.User.Id}/Images/Primary/?tag=${account.User.PrimaryImageTag}&quality=90`
+              : '/images/os_logo_square.png',
+          userType: UserType.JELLYFIN,
+        });
+        await userRepository.save(user);
+
+        //Update hostname in settings if it doesn't exist (initial configuration)
+        //Also set mediaservertype to JELLYFIN
+        if (settings.jellyfin.hostname == '') {
+          settings.main.mediaServerType = MediaServerType.JELLYFIN;
+          settings.jellyfin.hostname = body.hostname ?? '';
+          settings.save();
+        }
+      }
+    }
+
+    // Set logged in session
+    if (req.session) {
+      req.session.userId = user?.id;
+    }
+
+    return res.status(200).json(user?.filter() ?? {});
+  } catch (e) {
+    if (e.message != 'Unauthorized') {
+      logger.error(e.message, { label: 'Auth' });
+      return next({
+        status: 500,
+        message: 'Something went wrong. Is your auth token valid?',
+      });
+    } else {
+      return next({
+        status: 401,
+        message: 'CREDENTIAL_ERROR',
+      });
```

Comment:
We should probably log failed login attempts, like we do for failed local logins.

```suggestion
      logger.info('Failed login attempt from user with incorrect Jellyfin credentials', {
        label: 'Auth',
        account: {
          ip: req.ip,
          email: body.username,
          password: '__REDACTED__',
        },
      });
      return next({
        status: 403,
        message: 'You do not have access to this Jellyfin server',
      });
```